src/components/TopBar.jsx

```diff
@@ -0,0 +1,183 @@
+/* global _ */
+import React, { useEffect, useState, useRef } from 'react';
+import cx from 'classnames';
+import Telemetry from 'src/libs/telemetry';
+import Suggest from 'src/components/ui/Suggest';
+import Menu from 'src/panel/Menu';
+import { useConfig, useDevice } from 'src/hooks';
+import { handleFocus } from 'src/libs/input';
+import { selectItem, fetchSuggests } from 'src/libs/suggest';
+
+const MAPBOX_RESERVED_KEYS = ['ArrowLeft', 'ArrowUp', 'ArrowRight', 'ArrowDown', '-', '+', '='];
+
+const TopBar = ({
+  appInputValue,
+  userInputValue,
+  setUserInputValue,
+  inputRef,
+  onSuggestToggle,
+  backButtonAction,
+}) => {
+  const suggestElement = useRef(null);
+  const [focused, setFocused] = useState(false);
+  const { isMobile } = useDevice();
+  const config = useConfig();
+
+  // give keyboard focus to the field when typing anywhere
+  useEffect(() => {
+    const globalKeyHandler = e => {
+      if (MAPBOX_RESERVED_KEYS.find(key => key === e.key)) {
+        return;
+      }
+      // KeyboardEvent.key is either the printed character representation or a standard value for specials keys
+      // See https://developer.mozilla.org/fr/docs/Web/API/KeyboardEvent/key/Key_Values
+      if (
+        e.key.length === 1 &&
+        !e.ctrlKey &&
+        !e.metaKey &&
+        document.activeElement?.tagName !== 'INPUT'
+      ) {
+        inputRef.current.focus();
+      }
+    };
+
+    document.addEventListener('keydown', globalKeyHandler);
+
+    return () => {
+      document.removeEventListener('keydown', globalKeyHandler);
+    };
+  }, [inputRef]);
+
+  const onClickDirections = () => {
+    Telemetry.add(Telemetry.HOME_ITINERARY);
+    window.app.navigateTo('/routes');
+  };
+
+  const onSelectSuggestion = (item, query) => {
+    selectItem(item, { query });
+    inputRef.current.blur();
+  };
+
+  const onSubmit = async e => {
+    e.preventDefault();
+    Telemetry.add(Telemetry.SUGGEST_SUBMIT);
+    const query = inputRef.current.value;
+    const results = await fetchSuggests(query, {
+      withCategories: true,
+      useFocus: true,
+    });
+    onSelectSuggestion(results[0], {
+      query,
+      replaceUrl: true,
+    });
+  };
```

Comment:
There are some inconsistencies in the params passed to `onSelectSuggestion` and `selectItem`. 
According to line 56, the query should not be nested in a object, but the usage of `<Suggest onSubmit=...>` should be checked too.

As a consequence, an object may be used as the input value:
![image](https://user-images.githubusercontent.com/4726554/118610226-d4aff200-b7bb-11eb-8c35-c6bec021fce9.png)